mlflow/system_metrics/system_metrics_monitor.py

```diff
@@ -60,8 +69,26 @@ def __init__(self, run_id, sampling_interval=10, samples_before_logging=1):
         self.mlflow_logger = BatchMetricsLogger(self._run_id)
         self._shutdown_event = threading.Event()
         self._process = None
-        self._logging_step = 0
         self._metrics_prefix = "system/"
+        self._logging_step = self._get_next_logging_step(run_id) if resume_logging else 0
+
+    def _get_next_logging_step(self, run_id):
+        from mlflow.tracking.client import MlflowClient
+
+        client = MlflowClient()
+        try:
+            run = client.get_run(run_id)
+        except Exception:
```

Comment:
`Exception` seems too broad. Can we catch `MlflowException`?

Reply:
sg!